### PR TITLE
Create PlatformIO environments and update .gitignore

### PR DESCRIPTION
**Pull request by Burke-Daniel**
This Pull Request contains a platformio.ini file which contains an environment for each testing project, plus testing, and an example environment. Each project contains an environment for the arduino uno platform (board we will use for debugging code), and the SAMD21 chip architecture (sensor board chip). As well, the project .gitignore has been updated, and the `main.cpp` files in each project's directory have been changed to include a sample program to be used to test that the environments have been set up correctly.

### CODE REVIEW

File: Node/platformio.ini [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+[env:Test]
+platform = atmelavr
+board = uno
+framework = arduino
+src_filter = 
+    +<Test>
+build_flags = 
+    -I./Test/
+    -I./Common/drivers/inc
+    -I./Common/sensors/inc
```
Comment (colton-smith) [inline Node/platformio.ini]: Does the platform for testing have to be native?

File: Node/Test/test.cpp [hunk 2]
```diff
@@ -0,0 +1 @@
+// This file is intentionally left blank
```
Comment (colton-smith) [inline Node/Test/test.cpp]: Consider renaming the folder and env to `Testing` - test seems like it might be a one-off placeholder type of thing.

**Review by colton-smith — CHANGES_REQUESTED** [review]
Looks really nice, the only thing that's not concrete yet is the testing environment - you could add a testing example to ensure its working in this branch or set that up in another branch - totally up to you  👍 I think it's just the `platform` in the `Test` environment that might need to be changed, and possible the testing folder name.

I really like the skeleton for embedded!!

File: Node/platformio.ini (hunk 1 above)
Comment (Burke-Daniel) [reply Node/platformio.ini]: Yes, fixed and changed Test to Testing

File: Node/Test/test.cpp (hunk 2 above)
Comment (Burke-Daniel) [reply Node/Test/test.cpp]: Changed folder and env to Testing

**Review by colton-smith — APPROVED** [review]
GTM

**Pull request merged**